js/packages/nugget-logger/index.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs');
+
+const levels = [
+    'ERROR',
+    'WARN',
+    'INFO',
+    'VERBOSE',
+    'DEBUG',
+    'SILLY'
+];
+
+class logger {
+
+    /**
+     * @param level - name (not index) of the level for this logger to be
+     * @param logFilePath - file name/path for file to write logs to
+     */
+    constructor(level, logFilePath = '') {
+        // make sure the level is between 0 and 5
+        level = levels.indexOf(level.toUpperCase());
+        if (level < 0 || level > levels.length - 1)
+            throw 'level must be between 0 and 5 inclusice';
+
+        // this.level is set to a number instead of the actual name of the level
+        // the number is the index of the level name in `levels`
+        this.level = level;
+
+        this.logFile = logFilePath
+            ? fs.createWriteStream(logFilePath, { flags: 'a' })
+            : this.logFile = {
+                write: () => {}
+            };
+    }
+
+    print(label, message, level) {
+        // don't log anything above the level we specified in the constructor
+        if (level > this.level)
+            return;
+
+        const logMessage = `${new Date().toISOString()} ${levels[level]} [${label}] ${message}`;
+        console.log(logMessage);
+        this.logFile.write(logMessage + '\n');
+    }
+
+    e(label, message) { this.print(label, message, 0) };
+    w(label, message) { this.print(label, message, 1) };
+    i(label, message) { this.print(label, message, 2) };
+    v(label, message) { this.print(label, message, 3) };
+    d(label, message) { this.print(label, message, 4) };
+    s(label, message) { this.print(label, message, 5) };
```

Comment:
So I'm guessing `e` is for error and `w` is for warning, but what about the rest? Why do we want them?

Reply:
- `e`rror
- `w`arn
- `i`nfo
- `v`erbose
- `d`ebug
- `s`illy
Not gonna lie I copied this straight out of winston but winston was being a bitch so I just re-wrote it myself. This is just so we have options. Like that one log at the end of `remote/index.js` that would crash the program if you ran it locally, now that log is logged as `(v)erbose` and we can specify a lower max logging value than that so we can keep our console output clean.